helsingborg-stad/app-mitt-helsingborg
# Feat/form close dialog

## Explain the changes you’ve made

Adds a simple popup dialog window when the user exits the form. Styling is basic, Gustav can take a look at it and see what improvements can be done. 

## Explain your solution

This uses the Popup functionality from [previous PR](https://github.com/helsingborg-stad/app-mitt-helsingborg/pull/348), so that one should be reviewed and merged first. That's also why this PR appears bigger than it really is. I'm just writing this while it's fresh in my mind. 

I made a simple CloseDialog component, and changed Step so that it shows when the user tries to exit the form. This uses a TouchableNoFeedback to block the user from clicking on the form while the Popup shows, and a view with absolute positioning to put the popup in the middle of the screen. 

## How to test the changes?

Checkout this branch, and go into any form. When trying to exit, the popup should show up. Check that you cannot trigger things in the background while the popup is showing. 

## Was this feature tested in the following environments?
- [] Storybook on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

## Screenshots
![close dialog](https://user-images.githubusercontent.com/7421890/100751371-82cf0780-33e7-11eb-966e-8ffb385e40a5.png)
